dev/com.ibm.ws.kernel.feature.core/src/com/ibm/ws/kernel/feature/internal/subsystem/FeatureResourceImpl.java

```diff
@@ -373,4 +373,20 @@ public String getBundleRepositoryType() {
         return result;
     }
 
+    @Override
+    public String getRequiredOSGiEE() {
+        String result = _requiredOSGiEE.get();
+        if (result == null) {
+            // Directive names are in the attributes map, but end with a colon
+            result = _rawAttributes.get("required-osgi-ee:");
+            if (result == null)
+                result = "";
+            if (!_requiredOSGiEE.compareAndSet(null, result))
+                result = _requiredOSGiEE.get();
+        }
+        if ("".equals(result))
+            return null;
+        else
+            return result;
+    }
```

Comment:
This seems overly complicated to cache a map lookup of the key "required-osgi-ee:" into an atomic reference.  I think we should just go with a simple map lookup here:
```
    public String getRequiredOSGiEE() {
        // Directive names are in the attributes map, but end with a colon
        return _rawAttributes.get("required-osgi-ee:");
    }
```
The tolerates method does this caching to avoid splitting the raw string to a List<String> each call.  But here we just have a simple String so no need to have a complicated cache for that.

Reply:
I agree, but I wanted to follow the convention of the other attributes in this class:
```java
    private final AtomicReference<VersionRange> _range = new AtomicReference<VersionRange>();
    private final AtomicReference<String> _location = new AtomicReference<String>();
    private final AtomicReference<List<String>> _osList = new AtomicReference<List<String>>();
    private final AtomicReference<Map<String, String>> _attributes = new AtomicReference<Map<String, String>>();
    private final AtomicReference<Map<String, String>> _directives = new AtomicReference<Map<String, String>>();
    private final AtomicInteger _startLevel = new AtomicInteger(-1);
    private final AtomicReference<List<String>> _tolerates = new AtomicReference<List<String>>();
```
(Specifically, the `_location` attribute is what I followed the convention of since it is just a single String)

Reply:
I see, but I think the way `_location` does it is unnecessary also.  I would prefer not to follow that pattern if we think it is unnecessary work.